Sources/SwiftKafka/KafkaPollingSystem.swift

```diff
@@ -0,0 +1,299 @@
+        get {
+            self.stateMachineLock.withLockedValue { stateMachine in
+                return stateMachine.pollClosure
+            }
+        }
+        set {
+            self.stateMachineLock.withLockedValue { stateMachine in
+                stateMachine.pollClosure = newValue
+            }
+        }
+    }
+
+    /// Initializes the ``KafkaBackPressurePollingSystem``.
+    /// Private initializer. The ``KafkaBackPressurePollingSystem`` is not supposed to be initialized directly.
+    /// It must rather be initialized using the ``KafkaBackPressurePollingSystem.createSystemAndSequence`` function.
+    /* USE createSystemAndSequence, TESTING only */ init() {
+        self.stateMachineLock = NIOLockedValueBox(StateMachine())
+    }
+
+    /// Factory method creating a ``KafkaBackPressurePollingSystem`` and the ``AsyncSequence`` that receives its messages .
+    /// The caller of this function must retain the sequence in order to receive messages.
+    ///
+    /// - Returns: A tuple containing the ``KafkaBackPressurePollingSystem`` and a reference to the ``KafkaAsyncSequence``.
+    static func createSystemAndSequence(
+        backPressureStrategy: BackPressure
+    ) -> (KafkaPollingSystem, KafkaAsyncSequence<Element, BackPressure>) {
+        let pollingSystem = KafkaPollingSystem()
+
+        // (NIOAsyncSequenceProducer.makeSequence Documentation Excerpt)
+        // This method returns a struct containing a NIOAsyncSequenceProducer.Source and a NIOAsyncSequenceProducer.
+        // The source MUST be held by the caller and used to signal new elements or finish.
+        // The sequence MUST be passed to the actual consumer and MUST NOT be held by the caller.
+        // This is due to the fact that deiniting the sequence is used as part of a trigger to
+        // terminate the underlying source.
+        let acknowledgementsSourceAndSequence = NIOAsyncSequenceProducer.makeSequence(
+            elementType: Element.self,
+            backPressureStrategy: backPressureStrategy,
+            delegate: pollingSystem
+        )
+
+        pollingSystem.stateMachineLock.withLockedValue { stateMachine in
+            stateMachine.sequenceSource = acknowledgementsSourceAndSequence.source
+        }
+
+        let sequence = KafkaAsyncSequence(
+            wrappedSequence: acknowledgementsSourceAndSequence.sequence
+        )
+
+        return (pollingSystem, sequence)
+    }
+
+    /// Runs the poll loop with the specified poll interval.
+    ///
+    /// - Parameter pollInterval: The desired time interval between two consecutive polls.
+    /// - Returns: An awaitable task representing the execution of the poll loop.
+    func run(pollInterval: Duration) async {
+        self.stateMachineLock.withLockedValue { stateMachine in
+            guard stateMachine.running == false else {
+                fatalError("Poll loop must not be started more than once")
+            }
+            stateMachine.running = true
```

Comment:
This should be done inside a method of the state machine. Let's add a method `run()` to the state machine as well that returns a `RunAction`.